### ISSUE TITLE
Xcode 14.3 Beta Unable to compile with error 'MainActor' is only available in iOS 13.0 or newer

### ISSUE THREAD

**Issue by yishair-investing** [facebook/facebook-ios-sdk#2181]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-ios-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](https://github.com/facebook/facebook-ios-sdk/blob/main/CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Xcode version

14.3 Beta

### Facebook iOS SDK version

16.0.0

### Dependency Manager

SPM

### SDK Framework

Core

### Goals

Compile the SDK and App.

### Expected results

App should compile.

### Actual results

App fails to compile with error message: 

'MainActor' is only available in iOS 13.0 or newer

### Steps to reproduce

Install SDK and try compile the app. 

This could be related/duplicate https://github.com/facebook/facebook-ios-sdk/issues/2180

**Comment by marinofelipe** [comment]
Having the same issue here 👍 , whenever compiling with Swift 5.8's toolchain

**Comment by BoOelkers** [comment]
Same as well

**Comment by kielgillard** [comment]
My team has this same issue with Xcode 14.3, too.

**Comment by jordikitto** [comment]
Also seeing this with Xcode 14.3b1 - some other packages have already updated to handled this but not this package

**Comment by andre-alves** [comment]
Bump.. can't compile my app with Xcode 14.3 Beta and these frameworks:

```
FBSDKCoreKit
FBSDKLoginKit
FBSDKShareKit
```

**Comment by ecant** [comment]
Seeing this in our app too. Just tested with Xcode 14.3 Beta 2 and the issue is still present.

Also seeing this error message as a part of this, which breaks my understanding of [Module Stability](https://www.swift.org/blog/library-evolution/): 
```
Failed to build module 'FBSDKCoreKit'; this SDK is not supported by the 
compiler (the SDK is built with 'Apple Swift version 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51)', 
while this compiler is 'Apple Swift version 5.8 (swiftlang-5.8.0.119.11 clang-1403.0.22.11.100)'). 
Please select a toolchain which matches the SDK.
```

**Comment by kielgillard** [comment]
Can confirm, still present in beta 2.

**Comment by marinofelipe** [comment]
If only we could contribute to patching a fix.. but since it's an XCFramework that seems to use a remote source that we don't have access to 😅 🤷‍♂️ .. or perhaps we could fork and attempt to regenerate the local binary (though unsure precisely how the Facebook team does that), so...

Anyway, let us know if there's anything we can do to contribute towards having the fix earlier than later, Facebook team :)

**Comment by freddi-kit** [comment]
This would related to know issue: https://developer.apple.com/documentation/xcode-release-notes/xcode-14_3-release-notes#Known-Issues

**Comment by freddi-kit** [comment]
> The Xcode “Strict Concurrency Checking” setting is not correctly passed to the compiler. If you set it to “Minimal” (the default), the compiler will still perform “Targeted” checks. (105637789)

> Workaround: Add -Xfrontend -strict-concurrency=minimal to OTHER_SWIFT_FLAGS under your project’s Build Settings, or add -Xswiftc -strict-concurrency=minimal when running swift build.

**Comment by SunZhiC** [comment]
Follow Xcode alert, click fix button, the MainActor error should disappear.

**Comment by kielgillard** [comment]
This is still an issue in Xcode 14.3 beta 3.

It didn't seem like adding those compiler flags to my project did anything (I cannot see the compiler flags passed in the `SwiftEmitModule` step) https://github.com/facebook/facebook-ios-sdk/issues/2181#issuecomment-1455993067:
```
SwiftEmitModule normal arm64 Emitting\ module\ for\ FacebookLogin (in target 'FacebookLogin' from project 'Facebook')
    cd /REDACTED/SourcePackages/checkouts/facebook-ios-sdk
    builtin-swiftTaskExecution -- /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -emit-module -experimental-skip-non-inlinable-function-bodies-without-types /REDACTED/SourcePackages/checkouts/facebook-ios-sdk/Sources/FacebookLogin/Exports.swift -target arm64-apple-ios11.0-simulator -Xllvm -aarch64-use-tbi -enable-objc-interop -sdk /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk -I /REDACTED/Build/Products/Debug-iphonesimulator -I /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/lib -F /REDACTED/Build/Products/Debug-iphonesimulator/PackageFrameworks -F /REDACTED/Build/Products/Debug-iphonesimulator/PackageFrameworks -F /REDACTED/Build/Products/Debug-iphonesimulator -F /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks -F /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk/Developer/Library/Frameworks -no-color-diagnostics -enable-testing -g -module-cache-path /Users/kgillard/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -profile-generate -profile-coverage-mapping -swift-version 5 -enforce-exclusivity\=checked -Onone -D SWIFT_PACKAGE -D DEBUG -D Xcode -serialize-debugging-options -empty-abi-descriptor -Xcc -working-directory -Xcc /REDACTED/SourcePackages/checkouts/facebook-ios-sdk -resource-dir /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -enable-anonymous-context-mangled-names -Xcc -ivfsstatcache -Xcc /Users/kgillard/Library/Developer/Xcode/DerivedData/SDKStatCaches.noindex/iphonesimulator16.4-20E5238a-.sdkstatcache -Xcc -I/REDACTED/Build/Intermediates.noindex/Facebook.build/Debug-iphonesimulator/FacebookLogin.build/swift-overrides.hmap -Xcc -I/REDACTED/Build/Products/Debug-iphonesimulator/include -Xcc -I/REDACTED/Build/Intermediates.noindex/Facebook.build/Debug-iphonesimulator/FacebookLogin.build/DerivedSources-normal/arm64 -Xcc -I/REDACTED/Build/Intermediates.noindex/Facebook.build/Debug-iphonesimulator/FacebookLogin.build/DerivedSources/arm64 -Xcc -I/REDACTED/Build/Intermediates.noindex/Facebook.build/Debug-iphonesimulator/FacebookLogin.build/DerivedSources -Xcc -DSWIFT_PACKAGE -Xcc -DDEBUG\=1 -module-name FacebookLogin -disable-clang-spi -target-sdk-version 16.4 -target-sdk-name iphonesimulator16.4 -emit-module-doc-path /REDACTED/Build/Intermediates.noindex/Facebook.build/Debug-iphonesimulator/FacebookLogin.build/Objects-normal/arm64/FacebookLogin.swiftdoc -emit-module-source-info-path /REDACTED/Build/Intermediates.noindex/Facebook.build/Debug-iphonesimulator/FacebookLogin.build/Objects-normal/arm64/FacebookLogin.swiftsourceinfo -emit-objc-header-path /REDACTED/Build/Intermediates.noindex/Facebook.build/Debug-iphonesimulator/FacebookLogin.build/Objects-normal/arm64/FacebookLogin-Swift.h -serialize-diagnostics-path /REDACTED/Build/Intermediates.noindex/Facebook.build/Debug-iphonesimulator/FacebookLogin.build/Objects-normal/arm64/FacebookLogin-master-emit-module.dia -emit-dependencies-path /REDACTED/Build/Intermediates.noindex/Facebook.build/Debug-iphonesimulator/FacebookLogin.build/Objects-normal/arm64/FacebookLogin-master-emit-module.d -parse-as-library -o /REDACTED/Build/Intermediates.noindex/Facebook.build/Debug-iphonesimulator/FacebookLogin.build/Objects-normal/arm64/FacebookLogin.swiftmodule -emit-abi-descriptor-path /REDACTED/Build/Intermediates.noindex/Facebook.build/Debug-iphonesimulator/FacebookLogin.build/Objects-normal/arm64/FacebookLogin.abi.json
```

> Follow Xcode alert, click fix button, the MainActor error should disappear.

Which alert? Or is this what Facebook developer need to do to fix the `FacebookLogin` module?

**Comment by SajjadKharrazi** [comment]
Fixing with https://github.com/apple/swift/pull/64412.
tnx to @tshortli

**Comment by cameronehrlich** [comment]
Our team just ran into this with Xcode 14.3 RC using Xcode Cloud, hopefully @tshortli's fix makes it into the GM! 🤞

**Comment by leogiroux** [comment]
Our team just ran into this issue too, what can we do to solve it ? Our Xcode Cloud can't build testflight anymore.

**Comment by xmollv** [comment]
> Our team just ran into this with Xcode 14.3 RC using Xcode Cloud, hopefully @tshortli's fix makes it into the GM! 🤞

Yeah, we have the same issue with the RC in Xcode Cloud.

**Comment by leogiroux** [comment]
For xcode cloud just set manually the xcode version to 14.2, it works.
<img width="1104" alt="Capture d’écran 2023-03-22 à 13 51 28" src="https://user-images.githubusercontent.com/12122348/226909978-9e3dd59c-53ec-48df-87cc-7290f09a6e5f.png">

**Comment by jakesquareone** [comment]
XCode Cloud just stopped working for us too, what is the fix here @samodom?

**Comment by nirb1991** [comment]
I can see that it's already fixed [here](https://github.com/apple/swift/pull/64412), but I'm still getting the error in Xcode 14.3 RC.
Does the RC version contain the fix?

**Comment by raxityo** [comment]
This issue (along with #2180) has been fixed in Xcode 14.3 RC2. The app compiles and builds successfully for me 🎉 

From the [Relese Notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-14_3-release-notes#Resolved-Issues:~:text=were%20built%20with-,BUILD_LIBRARY_FOR_DISTRIBUTION,-enabled.%20These%20failures):
> Fixed: The Swift compiler may fail to build modules for XCFramework dependencies that were built with `BUILD_LIBRARY_FOR_DISTRIBUTION` enabled. These failures will occur when the XCFramework contains public Swift declarations that have `@MainActor` constraints implicitly added, such as subclasses of `UIView` or `NSView`. The Swift compiler expects those declarations to be marked `@available` for an operating system version where Swift concurrency is available.
> (105610970)
